src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3378,7 +3378,7 @@ private void AddCache(IEnumerable<NamespaceOrTypeSymbol> symbols)
                     }
 
                     var sourceNamespace = symbol as SourceNamespaceSymbol;
-                    if (sourceNamespace != null)
+                    if (sourceNamespace != null && !sourceNamespace.IsInjected)
```

Comment:
>!sourceNamespace.IsInjected [](start = 51, length = 27)

I can see how this might avoid a crash, but it doesn't look like doing nothing for an injected namespace in this class is the right thing to do. Will it ever return symbols for the injected namespaces and types? Also, it looks like the body of the previous ```if``` can still run into similar problem.

Reply:
I didn't quite understand. This method builds a cache from syntax/declaration to symbol, but we have a symbol without syntax/declaration. So I don't think we should add it to the cache. I think that's ok, since nobody could look it up in the cache anyways (there would be no key to use in `GetCachedSymbol`).

Regarding the second point: we never need to inject a namespace if one already exists. So you would not get an injected symbol in a MergedNamespaceSymbol.

---
In reply to: [215434995](https://github.com/dotnet/roslyn/pull/29180#discussion_r215434995) [](ancestors = 215434995)